.github/workflows/ci-build.yml

```diff
@@ -9,14 +10,100 @@ on:
+        run: mv icat-ansible/vault.yml icat-ansible/group_vars/all
+      - name: Replace default payara user with Actions user
+        run: |
+          sed -i -e "s/^payara_user: \"glassfish\"/payara_user: \"runner\"/" icat-ansible/group_vars/all/vars.yml
+      - name: Amending roles
+        run: |
+          sed -i 's/role: authn-uows-isis/role: authn-anon/' icat-ansible/icat-test-hosts.yml
+
+      # Force hostname to localhost - bug fix for previous ICAT Ansible issues on Actions
+      - name: Change hostname to localhost
+        run: sudo hostname -b localhost
+
+      # Create local instance of ICAT
+      - name: Run ICAT Ansible Playbook
+        run: |
+          ansible-playbook icat-ansible/icat-test-hosts.yml -i icat-ansible/hosts --vault-password-file icat-ansible/vault_pass.txt -vv
+
+      - name: Checkout datagateway-api
+        uses: actions/checkout@v2
+        with:
+          repository: ral-facilities/datagateway-api
+          path: datagateway-api
+
+      # DataGateway API file setup
+      - name: Create log file
+        run: touch logs.log
+      - name: Configure log file location
+        run: echo "`jq -r --arg REPO_DIR "$GITHUB_WORKSPACE/logs.log" \
+          '.log_location=$REPO_DIR' datagateway-api/datagateway_api/config.json.example`" > datagateway-api/datagateway_api/config.json.example
+      - name: Configure API backend
+        run: echo "`jq -r --arg BACKEND "python_icat" \
+          '.backend=$BACKEND' datagateway-api/datagateway_api/config.json.example`" > datagateway-api/datagateway_api/config.json.example
+      - name: Set debug mode to true
+        run: |
+          sed -i -e "s/\"debug_mode\": false,/\"debug_mode\": true,/" datagateway-api/datagateway_api/config.json.example
+      - name: Set log level to DEBUG
+        run: |
+          sed -i -e "s/\"log_level\": \"WARN\",/\"log_level\": \"DEBUG\",/" datagateway-api/datagateway_api/config.json.example
+      - name: Create config.json
+        run: cp datagateway-api/datagateway_api/config.json.example datagateway-api/datagateway_api/config.json
+      
+      # DataGateway API dependencies
+      - name: Install Poetry
+        run: pip install poetry
+      - name: Install dependencies
+        run: cd datagateway-api; poetry install; cd ..
+      - name: Add dummy data to icatdb
+        run: |
+          cd datagateway-api; poetry run python -m util.icat_db_generator; cd ..
+
+      - name: Start API
+        run: cd datagateway-api; nohup poetry run python -m datagateway_api.src.main > api-output.txt &
+      
+      # DataGateway setup
       - name: Checkout repo
         uses: actions/checkout@v2
       - name: Setup Node.js
         uses: actions/setup-node@v1
         with:
           node-version: '14.x'
+
       # Cache yarn dependencies/ restore the cached dependencies during future workflows
       - name: Get yarn cache directory path
         id: yarn-cache-dir-path
```

Comment:
I could only comment on a limited number of lines but come to think of it, I can't think of any reason why this is here. We don't need any ICAT stack setup for the regular old unit tests, right? There's no overlap with the e2e tests, is there?

Reply:
Umm I'm actually not sure, I think I just pasted in the setup just cause those red crosses are sad to look at. One would assume ICAT might not be needed though? I'll remove it and see what happens :)

Reply:
I've removed the ICAT and DataGateway API setup and it seemed all good as per https://github.com/ral-facilities/datagateway/actions/runs/1049320004